core/matrix/csr.cpp

```diff
@@ -128,7 +127,7 @@ void Csr<ValueType, IndexType>::move_to(Dense<ValueType> *result)
 {
     auto exec = this->get_executor();
     auto tmp = Dense<ValueType>::create(exec, this->get_size());
-    exec->run(csr::make_move_to_dense(tmp.get(), this));
+    exec->run(csr::make_convert_to_dense(tmp.get(), this));
```

Comment:
Is there a possibility that this could be replaced by an actual `make_move_to_dense` operation ? Then maybe you should add a note here that says TODO or something like that.

Reply:
I think this is very unlikely since you can't really reuse any of the memory that you already allocated.
Currently, I think we only reuse part of the the memory for the conversion Csr <-> Coo, the rest actually calls `convert_to`.
I could (and I think should) actually directly call the `convert_to` method, which saves quite some duplicated code.